HIPs/firefly.md

```diff
@@ -0,0 +1,378 @@
+
+Enterprise blockchains are purpose built to deliver (1) with invaluable attributes like _global ordering_ and _finality_. FireFly will support Hyperledger Fabric, Enterprise Ethereum (such as Quorum and Besu), and will investigate Corda support. (1) also includes global storage providers such as IPFS. FireFly also provides a significant amount of technology for (2) and (3) including cross-party messaging, cross-party private document transfer, and event propagation. All of these runtimes and connectors into the FireFly node. For (4), Hyperledger FireFly also provides a network map and registry that unifies all of the identity and bindings information needed.
+
+## Motivation
+
+When building a system for multiparty business transactions based on blockchain technology, there are a number of problems that need to be solved. Problems such as:
+
+- How do parties agree on standardized data types?
+- How do I securely send private data between two parties in the system without other parties being able to view the data?
+- How do I run business processes in response to blockchain transactions?
+- How do I leverage blockchain technology, but not expose sensitive data stored on-chain?
+- How do I process events originated from multiple parties in the correct order that is uniformly honored throughout the decentralized system?
+
+Blockchain plays a key role in the solution for each of them, but it's not always obvious how to best utilize blockchain, or the solution can be quite complex. Today, there is no comprehensive OSS project that makes it easier and more reliable to solves these common problems. FireFly fills this gap, and allows businesses to focus on solving business problems.
+
+![Typical Blockchain Stack](../assets/stack.svg "Typical Blockchain Stack")
+
+The gap includes functionality such as registering and verifying member identity within the network (note that member identities needed in the app layer are not always onchain member identities defined in the blockchain layer, consider multi-tenant blockchain nodes), defining custom data types, sharing private or public data with network members (especially large file based data exchanges not suited for state DB storage), hashing and pinning data to the blockchain, securely sending events to network members, and triggering business process flows in response to events (typically involves enterprise grade messaging queue which is known to be difficult to integrate with).
+
+Because there is no existing OSS project that solves these problems, developers are left to themselves to create a solution. These solutions likely end up becoming specialized or tightly coupled to the project they were designed as a part of, and therefore are not reusable. Firefly seeks to fill this gap with a platform that is reusable for any multiparty business transaction backed by blockchain.
+
+## Status
+
+Proposal
```

Comment:
Should we update this to say `Incubation` now? Or leave it as is, since this document was the proposal?

Reply:
Good question!

[personal opinion]
This document is serving two purposes at once. While it has been very helpful for TSC to understand the project, it will also be put up in the HIP process website https://hyperledger.github.io/hyperledger-hip/HIPs/. It shall always reflect the current status. @ryjones here is another opportunity to make use of landscape and automation.